fem/bilininteg.cpp

```diff
@@ -2247,29 +2281,43 @@ void VectorDiffusionIntegrator::AssembleElementVector(
    if (ir == NULL)
    {
       // integrand is rational function if det(J) is not constant
-      int order = 2 * Tr.OrderGrad(&el); // order of the numerator
+      const int order = 2 * Tr.OrderGrad(&el); // order of the numerator
       ir = (el.Space() == FunctionSpace::rQk) ?
            &RefinedIntRules.Get(el.GetGeomType(), order) :
            &IntRules.Get(el.GetGeomType(), order);
    }
 
+   // If vdim is not set, set it to the space dimension;
+   vdim = (vdim <= 0) ? Tr.GetSpaceDim() : vdim;
+
    elvect = 0.0;
    for (int i = 0; i < ir->GetNPoints(); i++)
    {
       const IntegrationPoint &ip = ir->IntPoint(i);
+      el.CalcDShape(ip, dshape);
 
       Tr.SetIntPoint(&ip);
       CalcAdjugate(Tr.Jacobian(), Jinv);
-      w = ip.weight / Tr.Weight();
-      if (Q)
+      double w{ip.weight / Tr.Weight()};
+
+      if (VQ)
       {
-         w *= Q->Eval(Tr, ip);
+         mfem_error("VectorDiffusionIntegrator::AssembleElementVector \n"
```

Comment:
Why raising en error? can't you provide the application as done for the matrix assembly?

Reply:
I don't understand what this routine is supposed to do, I don't need it for my work, and there are no examples in any of examples, miniapps, or unittests. I'll try to implement it but I have no way of testing.

Reply:
`AssembleElementVector` is the routine that computes the FEM residual. For linear problems, 
```
elvect = elmat * elfun
```
 where `elfun` is the input linearization point .

Reply:
I've implemented with a different style than the original function but in the same style as `AssembleElementMatrix`.